src/main/scala/tech/sourced/gemini/FeaturesHash.scala

```diff
@@ -0,0 +1,62 @@
+package tech.sourced.gemini
+
+import tech.sourced.featurext.generated.service.Feature
+
+import scala.collection.mutable
+
+object FeaturesHash {
+  // All params below must be the same for hash & query
+  val defaultSeed = 0
+  val defaultSampleSize = 128
+  // Depend on threshold and sample_size
+  // this values calculated for threshold=0.8 and sample_size=128
+  val defaultHashtablesNum = 9
+  val defaultBandSize = 13
+
+  private var tokensSize: Int = _
+  private var wmh: WeightedMinHash = _
+
+  def hashFeatures(
+                    docFreq: OrderedDocFreq,
+                    features: Iterable[Feature],
+                    sampleSize: Int = defaultSampleSize,
+                    seed: Int = defaultSeed): Array[Array[Long]] = {
+    val OrderedDocFreq(docs, tokens, df) = docFreq
+
+    if (wmh == null || tokensSize != tokens.size) {
+      wmh = new WeightedMinHash(tokens.size, sampleSize, seed)
+    }
+
+    val bag = mutable.ArrayBuffer.fill(tokens.size)(0.toDouble)
```

Comment:
For me, ability to understand this logic would increase, if extracted to a method with descriptive name and possibly a link to analog in Apollo codebase in ScalaDoc comment.

Reply:
could you help me to write comment/change name?

The method has name `hashFeatures` it's exactly what this code is doing.
I also tried to write it not short but understandable giving variable meaningful names like bag/tf and so on. So I don't really see how to improve it by myself.

The code in apollo: https://github.com/src-d/apollo/blob/master/apollo/hasher.py#L204
but I personally get confused by it.